File: src/js/17_slowniki/zad9.js
/*


Tytul: Powtarzajace sie znaki

Otrzymujesz napis. Twoim zadaniem jest znalezienie i zwrocenie wszystkich znakow, ktore wystepuja co najmniej 2 razy w ciagu.

Dane wejsciowe: Napis.

Dane wyjsciowe: Napis.

Przyklad:

Dla otrzymanego napisu: "aaabbbccc"

Powinien zostac zwrocony napis: "abc"
*/

function powtarzajaceSieZnaki(napis) {
  const znaki = {};
  let powtarzajaceSie = "";
  for (let i = 0; i < napis.length; i++) {
    const znak = napis[i];
    if (!znaki[znak]) {
      znaki[znak] = 1;
    } else {
      znaki[znak]++;
    }
  }
  for (const znak in znaki) {
    if (znaki[znak] >= 2) {
      powtarzajaceSie += znak;
    }
  }
  return powtarzajaceSie;
}

function testPowtarzajaceSieZnaki() {
  const napis1 = "aaabbbccc";
  const wynik1 = powtarzajaceSieZnaki(napis1);
  console.assert(wynik1 === "abc", "Test 1 nie powiodl sie");

  const napis2 = "abcd";
  const wynik2 = powtarzajaceSieZnaki(napis2);
  console.assert(wynik2 === "", "Test 2 nie powiodl sie");

  const napis3 = "Ala ma kota, a kot ma Ale.";
  const wynik3 = powtarzajaceSieZnaki(napis3);
  console.assert(wynik3 === "aekmt", "Test 3 nie powiodl sie");
}

testPowtarzajaceSieZnaki();

File: src/python/17_slowniki/zad9.py
"""
Tytul: Czy slowniki sa identyczne?
Otrzymujesz dwa slowniki skladajace sie z par napisow i list liczb calkowitych. Listy sa nieuporzadkowane. Twoim zadaniem jest sprawdzenie, czy slowniki skladaja sie z tych samych par. Dwie listy uznajemy za identyczne, jesli zawieraja te same elementy, niezaleznie od kolejnosci, w jakiej sie one znajduja.
Dane wejsciowe: Dwa slowniki par: napis, lista liczb calkowitych.
Dane wyjsciowe: Wartosc boolowska.
Przyklad:
Dla otrzymanych slownikow:
{'a': [1, 2, 3], 'b': [4, 5]}, {'a': [3, 2, 1], 'b': [5, 4]}
Powinna zostac zwrocona wartosc logiczna: Prawda.
Dla otrzymanych slownikow:
{'a': [1, 2, 3], 'b': [4, 5]}, {'a': [3, 2, 1], 'b': [5, 4], 'c': [6, 7]}
Powinna zostac zwrocona wartosc logiczna: Falsz.

"""


def znaki_powtarzajace_sie_wiecej_niz_raz(napis):
    """
    Funkcja zwraca liste znakow powtarzajacych sie wiecej niz raz.
    """
    znaki = []
    for znak in napis:
        if napis.count(znak) > 1:
            znaki.append(znak)
    return znaki


def test_znaki_powtarzajace_sie_wiecej_niz_raz():
    assert znaki_powtarzajace_sie_wiecej_niz_raz("ababab") == ["a", "b"]
    assert znaki_powtarzajace_sie_wiecej_niz_raz("abc") == []


if __name__ == "__main__":

    test_znaki_powtarzajace_sie_wiecej_niz_raz()

